cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunTestsTask.java

```diff
@@ -311,6 +327,57 @@ private int runTestSuite(Target target, Package currentPackage, JBallerinaBacken
         return proc.waitFor();
     }
 
+    private void getclassFromSourceFilePath(List<String> sourceFileList, Package currentPackage,
+                                                                            List<String> classFileList) {
+        String sourceRoot = currentPackage.project().sourceRoot().toString();
+        for (String sourceFile : sourceFileList) {
+            validateSourceFilePath(sourceRoot, sourceFile);
+            if (sourceFile.startsWith(sourceRoot)) {
+                sourceFile = sourceFile.replace(sourceRoot + PATH_SEPARATOR, "");
+            } else if (sourceFile.startsWith(RELATIVE_PATH_PREFIX)) {
+                sourceFile = sourceFile.replace(RELATIVE_PATH_PREFIX, "");
+            }
+            String org = currentPackage.packageOrg().toString();
+            String packageName = currentPackage.packageName().toString();
+            String version = currentPackage.packageVersion().toString();
+
+            if (sourceFile.contains(("modules")) || sourceFile.contains(("generated"))) {
```

Comment:
Shall we add constants for `modules` and `generated`? Let's check if we can reuse any existing constants.

Reply:
Let's add the usage and some examples in the PR description too.